lib/logstash/config/defaults.rb

```diff
@@ -0,0 +1,9 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "concurrent"
+
+module LogStash module Config module Defaults
+  def self.cpu_cores
+    Concurrent.processor_count
```

Comment:
I suppose we could just note (comment it) what processor count is returned (physical cores vs hyperthreads) and that would satisfy me :)